core/baseline-core-v1.0-psd01.md

```diff
@@ -2786,113 +2795,722 @@ The document defines storage capability requirements in the following areas:
+
+Test steps:
+
+1. Create a BPI Subject within the BPI with a public-private cryptographic key pair based on one of the cryptographic algorithms supported by the authentication service chosen for the BPI Subject
+2. Create an authentication policy for the BPI Storage supported by each of the authemtication services chosen for the test  
+2. Attempt to authenticate the BPI Subject with the BPI Storage using each of the identified external authentication services for the test.
+3. Verify that the authentication of the BPI Subject is successful and the BPI Subject is granted the appropriate access to the BPI Storage as per the test authentication policy.
+4. Repeat the above steps for each supported external authentication service.
+
+Test pass criteria:
+
+The test will pass if one or more of the external authentication services successfully authenticate a BPI Subject to the BPI Storage and the BPI Subject is granted the type of access to the BPI Storage as defined in the test policy. 
+
+Note that the test policy could be any combination of read and write access to the BPI Storage.
+
 #### **[R283]**	
 BPI Storage MUST support roles & access management.
 
+[[R283]](#r283) Testability: 
+
+Preconditions: 
+
+* Identify the roles that need to be supported by the BPI Storage, based on the functional requirements and design specifications such as "Administrators" with full read and write permissions or "ReadUser" with only read permissions.
+
+Test steps:
+
+1. Create two or mor test BPU Subjects and assign them to the identified roles.
+2. Attempt to perform various operations on the BPI Storage using each test BPI Subject such as read and write operations.
+3. Verify that the operations are allowed or denied based on the permissions associated with the BPI Subject's role.
+4. Attempt to perform operations that require higher permissions than the current BPI Subject's role such as writing for a "ReadUser" role.
+5. Verify that such operations are denied and an appropriate error message is displayed.
+6. Repeat the above steps for each supported role.
+
+Test pass criteria:
+
+The test will pass if all supported roles are successfully tested, and BPI Subjects are able to perform operations only as allowed by their role's permissions. Additionally, attempting operations that require higher permissions than the current BPI Subject's role must be denied, and appropriate error messages must be displayed.
+
 #### **[R284]**	
 BPI Storage MUST support policy management.
 
+[[R284]](#r284) Testability:
+
+Preconditions:
+
+* A BPI test environment is installed and configured properly.
+* The BPI storage component is designed to support policy management.
+* A BPI Subject has appropriate permissions to create, modify, and delete policies.
+
+Test Steps:
+
+1. Create a set of policies that the BPI storage component should be able to manage, including access control policies, retention policies, and deletion policies.
```

Comment:
As with comments for R283, the requirement seems underspecified. If the BPI has to support managing policies for
- access control,
- retention, and
- deletion,

that should be written into the requirement itself.

Reply:
added policy examples and removed the specific example from the test and modified the language to be more concise and precise without requiring specific examples. The requirement is not under-specified. It is intentionally left very broad leaving significant implementation flexibility.

Reply:
The requirement is broad, but I think it is also under-specified. Implementing RBAC or some other access control policy likely meets it, making the requirement redundant as is - but equally, as written, implementing management of whether to use stripes or not in reporting outputs could be argued as meeting it. (Admittedly that's a pretty perverse argument).